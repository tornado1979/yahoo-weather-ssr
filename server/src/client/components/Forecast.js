import React from 'react'
import propTypes from 'prop-types'

export const Forecast = ({ forecast }) => {
  return (
    <div className="forecast-item">
      <div className="day">{forecast.day}</div>
      <div className="temp-container">
        <div className="low-temp">
          <span>L</span>
          {forecast.low}
        </div>
        <div className="high-temp">
          <span>H</span>
          { forecast.high }
        </div>
      </div>
    </div>
  )
}

Forecast.propTypes = {
  forecast: propTypes.shape({
    day: propTypes.string,
    hight: propTypes.string,
    low: propTypes.string,
  }).isRequired,
}
