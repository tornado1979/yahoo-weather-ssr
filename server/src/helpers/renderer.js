import React from 'react'
import { renderToString } from 'react-dom/server'
import { StaticRouter } from 'react-router-dom'
import { Provider } from 'react-redux'
import { renderRoutes } from 'react-router-config'
import serialize from 'serialize-javascript'

import Routes from '../client/Routes'

export default (req, store, context) => {
  const content = renderToString(
    <Provider store={store}>
      <StaticRouter context={context} location={req.path}>
        <div>{renderRoutes(Routes)}</div>
      </StaticRouter>
    </Provider>,
  )
  return `
  <!DOCTYPE html>
  <html>
    <head>
      <meta charset="UTF-8">
      <title>Server Side render application</title>
    </head>
   <body>
     <div id="root">${content}</div>
     <script>
       window.INITIAL_STATE = ${serialize(store.getState())}
     </script>
     <script src="bundle.js"></script>
   </body>
  </html>
  `
}
