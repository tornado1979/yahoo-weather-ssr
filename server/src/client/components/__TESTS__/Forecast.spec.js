import React from 'react'
import renderer from 'react-test-renderer'

import { Forecast } from '../Forecast'

describe('Component Forecast', () => {
  it('renders correctly', () => {
    const forecast = {
      day: 'Monday',
      low: '20',
      high: '30',
    }

    const tree = renderer
      .create(<Forecast forecast={forecast} />)
      .toJSON()

    expect(tree).toMatchSnapshot()
  })
})
