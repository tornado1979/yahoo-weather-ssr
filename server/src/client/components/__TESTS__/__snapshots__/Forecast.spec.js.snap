// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Forecast renders correctly 1`] = `
<div
  className="forecast-item"
>
  <div
    className="day"
  >
    Monday
  </div>
  <div
    className="temp-container"
  >
    <div
      className="low-temp"
    >
      <span>
        L
      </span>
      20
    </div>
    <div
      className="high-temp"
    >
      <span>
        H
      </span>
      30
    </div>
  </div>
</div>
`;
